js/bundle/src/constants.ts

```diff
@@ -0,0 +1,4 @@
+export const B1 = 'b1';
+export const B2 = 'b2';
```

Comment:
I am not sure about the value of the `B1` and `B2` constants - I think these could just as well remain strings.

Reply:
I think the value is if someone decides to refactor them, they cant do it without changing all the places and breaking the code. With strings it's possible to leave something unchanged and not notice it since the code would still compile

Reply:
> With strings it's possible to leave something unchanged and not notice it since the code would still compile

My goal would be to make things typesafe-enough that it would no longer compile if that happens. But I do understand that this might be out of scope of this PR.